core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -677,11 +677,18 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
     }
   }
 
-  def activeProducers = pendingTransactionalOffsetCommits.keySet
+  def activeProducers: collection.Set[Long] = pendingTransactionalOffsetCommits.keySet
 
-  def hasPendingOffsetCommitsFromProducer(producerId: Long) =
+  def hasPendingOffsetCommitsFromProducer(producerId: Long): Boolean =
     pendingTransactionalOffsetCommits.contains(producerId)
 
+  def hasPendingOffsetCommitsForTopicPartition(topicPartition: TopicPartition): Boolean = {
```

Comment:
Leaving comment here for lack of alternative. We had discussed having the coordinator enforce exclusive access to a member for committing a given partition. Did we decide this was unnecessary?

Reply:
https://issues.apache.org/jira/browse/KAFKA-9421 Get a JIRA to track it. At the moment it doesn’t seem necessary for existing use cases.
Also for really advanced users, they may actually want to let their consumers work concurrently on one partition,
I guess this will then become a user’s responsibility for making their customized partition assignor behave correctly.